dateparser/custom_language_detection/langdetect.py

```diff
@@ -0,0 +1,36 @@
+from langdetect.detector_factory import DetectorFactory, PROFILES_DIRECTORY
+import langdetect
+
+
+# The below Factory is set to prevent setting global state of the library
+# but still getting consistent results.
+# Refer - https://github.com/Mimino666/langdetect
+
+class Factory:
+    data = None
+
+
+def init_factory():
+    if Factory.data is None:
+        Factory.data = DetectorFactory()
+        Factory.data.load_profile(PROFILES_DIRECTORY)
+        Factory.data.seed = 0
+
+
+def detect_langs(text):
+    init_factory()
+    detector = Factory.data.create()
+    detector.append(text)
+    return detector.get_probabilities()
+
+
+def detect_languages(text, confidence_threshold):
+    language_codes = []
+    try:
+        parser_data = detect_langs(text)
+        for langauge_candidate in parser_data:
+            if langauge_candidate.prob > confidence_threshold:
+                language_codes.append(langauge_candidate.lang)
+    except langdetect.lang_detect_exception.LangDetectException:
+        pass
```

Comment:
This will by pass the exception silently. I think at least a log entry would be good.

Reply:
Hi @kishan3, this error is thrown when empty text is passed or no similar language (And this library only supports 55 languages) is found which is too often the case so I did not put any logging as that may create many such logs.

Reply:
+1 to ignore the exception - as it happens not only on empty strings but on inputs like ``10-10-2021`` which have no letters - which would be extremely typical to dateparser.

Reply:
Thanks for the review @lopuhin, so I will not change anything here

Reply:
Thanks, Marc, I have added these comments